graphql-dgs/src/main/java/com/netflix/graphql/dgs/internal/VirtualThreadTaskExecutor.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2023 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs.internal;
+
+import org.jetbrains.annotations.NotNull;
+import org.springframework.core.task.AsyncTaskExecutor;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.Future;
+
+/**
+ * Unimplemented version of the VirtualThreadTaskExecutor just for reachability.
```

Comment:
What happens if you don't provide this? Is it causing a missing Bean?

Reply:
No, it's injected here as an `Optional`, and the other code all checks if it's available or not. 

```
open fun methodDataFetcherFactory(argumentResolvers: ObjectProvider<ArgumentResolver>, @Qualifier("dgsAsyncTaskExecutor") taskExecutorOptional: Optional<AsyncTaskExecutor>
```

Reply:
Because it's also conditional on Java 21, so it needs to be able to handle the case where it's not there.